pinc/ApiClientStorage.inc

```diff
@@ -0,0 +1,38 @@
+<?php
+
+class ApiClientStorage
+{
+    private JsonStorage $storage;
+    private string $setting;
+
+    public function __construct(string $client, string $username)
+    {
+        global $api_client_storage_keys;
+
+        if (!in_array($client, $api_client_storage_keys)) {
```

Comment:
Would the client not already have been validated before we get here?

Reply:
It would. I did it here too to ensure it's enforced at the data layer not just the user input layer if this is used elsewhere for some reason.